modules/ztp-syncing-new-configmap-changes-to-existing-pgt-crs.adoc

```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc
+
+:_content-type: PROCEDURE
+[id="ztp-syncing-new-configmap-changes-to-existing-pgt-crs_{context}"]
+= Syncing new ConfigMap changes to existing PolicyGenTemplate CRs
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have logged in to the hub cluster as a user with `cluster-admin` privileges.
+
+* You have created a `PolicyGenTemplate` CR that pulls information from a `ConfigMap` CR using hub cluster templates.
+
+.Procedure
+
+. Update the contents of your `ConfigMap` CR, and apply the changes in the hub cluster.
+
+. To sync the contents of the updated `ConfigMap` CR to the deployed policy, do either of the following:
+
+.. Optional: Delete the existing policy and recreate it using the `PolicyGenTemplate` CR.
+
+... Delete the policy, for example:
++
+[source,terminal]
+----
+$ oc delete policy <policy_name> -n <policy_namespace>
+----
++
+... Re-apply the `PolicyGenTemplate` CR:
++
+[source,terminal]
+----
+$ oc apply -f sriov-pgt-example.yaml
+----
+
+.. Optional: Apply a special annotation `policy.open-cluster-management.io/trigger-update` to the policy with a different value every time when updating the `ConfigMap`. For example, run the following command:
++
+[source,terminal]
+----
+$ $ oc label <policy_name> -n <policy_namespace> policy.open-cluster-management.io/trigger-update="1"
+----
+
+... Re-apply the `PolicyGenTemplate` CR to regenerate the `ClusterGroupUpdate` CR. For example, run the following command:
++
+[source,terminal]
+----
+$ oc apply -f sriov-pgt-example.yaml
+----
+
+... If it exists, delete the existing `ClusterGroupUpdate` CR that contains the policy. Create a new `ClusterGroupUpdate` CR to apply the updated `ConfigMap` changes. For example:
++
+[source,terminal]
+----
+$ oc delete clustergroupupgrade <cgu_name> -n <cgu_namespace>
+----
```

Comment:
This should be step3.
```suggestion
.  Delete the existing `ClusterGroupUpdate` CR that contains the policy if it exists (optional). Create a new `ClusterGroupUpdate` CR with the policy included to apply the updated `ConfigMap` changes. For example:
+
[source,terminal]
----
$ oc delete clustergroupupgrade <cgu_name> -n <cgu_namespace>
$ oc apply -f clustergroupupgrade.yaml
----
```

Reply:
Updated. Could you give me an representative example CGR (`clustergroupupgrade.yaml`) for this please @Missxiaoguo.

Reply:
@Missxiaoguo I added an example CGU CR. PTAL.

Reply:
Question: do we need to include the list of clusters in the updated CGU CR?
```
spec:
  managedPolicies:
    - <managed_policy>
  enable: false
  clusters: <1>
  - <managed_cluster_1>
  - <managed_cluster_2>
```

Reply:
@Missxiaoguo does this look OK?

Reply:
Set `enable: true`, others look good.

Reply:
updated.